.github/workflows/pr-test-codegen-downstream.yml

```diff
@@ -0,0 +1,139 @@
+            number=$(gh search prs --repo pulumi/pulumi-${{ matrix.provider }} --review-requested ${{ toJSON( github.triggering_actor || 'justinvp' ) }} --match body "This is a downstream codegen test for pulumi/pulumi#${{ github.event.pull_request.number }}. (run-id: ${{ github.run_id }})"  --json number --jq '.[0].number')
+            # Ensure that expected checks are pending before checking the status passes.
+            # We use 'test' as all workflows have tests.
+            until gh pr checks --repo "pulumi/pulumi-${{ matrix.provider }}" "$number" | grep 'test'; do sleep 30; done;
+            gh pr checks --repo "pulumi/pulumi-${{ matrix.provider }}" "$number" --watch --fail-fast
+# Native Providers
+    awsx:
+      name: Test AWSX
+      needs: ["cleanup"]
+      timeout-minutes: 240
+      runs-on: ubuntu-latest
+      env:
+        GOVERSION: ">=1.19.0" # from awsx: decoupled from version sets, track latest for codegen
+        NODEVERSION: "14.x"
+        PYTHONVERSION: "3.9.x"
+        DOTNETVERSION: "6.x"
+        JAVAVERSION: "11"
+        AWS_REGION: us-west-2
+        GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+      steps:
+        - name: Configure AWS Credentials
+          uses: aws-actions/configure-aws-credentials@v1
+          with:
+            aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+            aws-region: ${{ env.AWS_REGION }}
+            aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            role-duration-seconds: 3600
+            role-session-name: awsx@githubActions
+            role-to-assume: ${{ secrets.AWS_CI_ROLE_ARN }}
+        - name: Setup Go
+          uses: actions/setup-go@v2
+          with:
+            go-version: ${{ env.GOVERSION }}
+            check-latest: true
+        - name: Setup Node
+          uses: actions/setup-node@v1
+          with:
+            node-version: ${{ env.NODEVERSION }}
+        - name: Setup Python
+          uses: actions/setup-python@v3
+          with:
+            python-version: ${{ env.PYTHONVERSION }}
+        - name: Setup DotNet
+          uses: actions/setup-dotnet@v1
+          with:
+            dotnet-version: ${{ env.DOTNETVERSION }}
+        - name: Setup Java
+          uses: actions/setup-java@v3
+          with:
+            distribution: temurin
+            java-version: ${{ env.JAVAVERSION }}
+        - name: Install gotestfmt
+          uses: jaxxstorm/action-install-gh-release@v1.7.1
+          with:
+            repo: gotesttools/gotestfmt
+        - name: Install pulumictl
+          uses: jaxxstorm/action-install-gh-release@v1.1.0
+          with:
+            repo: pulumi/pulumictl
+        - name: Check out source code
+          uses: actions/checkout@v3
+          with:
+            ref: ${{ github.event.pull_request.head.sha }}
+            token: ${{ secrets.PULUMI_BOT_TOKEN }}
+        - name: Test Downstream
+          uses: pulumi/action-test-provider-downstream@v0.0.1-beta
```

Comment:
Depends on a custom tagged version of this action that allows me to target specific `make` targets.